poetry/spdx/__init__.py

```diff
@@ -17,7 +17,36 @@ def license_by_id(identifier):
     id = identifier.lower()
 
     if id not in _licenses:
-        raise ValueError("Invalid license id: {}".format(identifier))
+        err = "Invalid license id: {}\nPoetry uses SPDX license identifiers: https://spdx.org/licenses/".format(
+            identifier
+        )
+
+        # Covers the licenses listed as common for python packages in https://snyk.io/blog/over-10-of-python-packages-on-pypi-are-distributed-without-any-license/
+        # MIT/WTFPL/Unlicense are excluded as their ids are simply their name - if someone types "mit", they've already found the license they were looking for
+
+        common_strings = ["agpl", "lgpl", "gpl", "bsd", "apache", "mpl", "cc0"]
+        for str in common_strings:
```

Comment:
```suggestion
        for string in common_strings:
```

`str` is a key word in Python and we should avoid using it as a variable name. I know `id` and `license` are too but they are less likely to be used compared to `str`.